app/bot.go

```diff
@@ -90,13 +93,13 @@ func (a *App) EnsureBot(c request.CTX, productID string, bot *model.Bot) (string
 		return user.Id, nil
 	}
 
-	createdBot, err := a.CreateBot(c, bot)
-	if err != nil {
-		return "", fmt.Errorf("failed to create bot: %w", err)
+	createdBot, cErr := a.CreateBot(c, bot)
+	if cErr != nil {
+		return "", fmt.Errorf("failed to create bot: %w", cErr)
 	}
 
 	if appErr := a.SetPluginKey(productID, botUserKey, []byte(createdBot.UserId)); appErr != nil {
-		return "", fmt.Errorf("failed to set plugin key: %w", err)
+		return "", fmt.Errorf("failed to set plugin key: %w", appErr)
```

Comment:
I got errors while linting because of `shadow: declaration`. After renaming the other variables which have been named just `err`, this actually revealed, that this should be `appErr`. Otherwise `err` is always `nil`.

Reply:
Same here: good catch :+1: